tests/ha/single_pod_restart/test_data_pod_restart_api.py

```diff
@@ -1221,3 +1222,249 @@ def test_obj_ver_suspension_data_pod_restart(self):
+        LOGGER.info("Step 1: Create bucket %s and perform upload of object size %s MB and "
+                    "Overwrite the object", self.bucket_name, file_size)
+
+        bkt_cnt = HA_CFG["copy_obj_data"]["bkt_multi"]
+        before_bkt = dict()
+        if CMN_CFG["dtm0_disabled"]:
+            LOGGER.info("Create %s buckets and upload objects for background overwrite during "
+                        "pod restart.", bkt_cnt)
+            t_t = int(perf_counter_ns())
+            for cnt in range(bkt_cnt):
+                before_bkt.update({f"ha-bkt{cnt}-{t_t}": [f"ha-obj{cnt}-{t_t}", file_size]})
+            resp = self.ha_api.object_overwrite_dnld(self.s3_test_obj, before_bkt, iteration=0,
+                                                     random_size=False)
+            assert_utils.assert_true(resp[0], "Failure observed in create new bucket, "
+                                              "upload object.")
+
+        LOGGER.info("Step 2: Shutdown data pod with replica method and verify cluster & "
+                    "remaining pods status")
+        resp = self.ha_obj.delete_kpod_with_shutdown_methods(
+            master_node_obj=self.node_master_list[0], health_obj=self.hlth_master_list[0],
+            delete_pod=[self.delete_pod], num_replica=self.num_replica - 1)
+        # Assert if empty dictionary
+        assert_utils.assert_true(resp[1], "Failed to shutdown/delete data pod")
+        pod_name = list(resp[1].keys())[0]
+        self.set_name = resp[1][pod_name]['deployment_name']
+        self.restore_pod = True
+        self.restore_method = resp[1][pod_name]['method']
+        assert_utils.assert_true(resp[0], "Cluster/Services status is not as expected")
+        LOGGER.info("Step 2: Successfully shutdown data pod %s. Verified cluster and "
+                    "services states are as expected & remaining pods status is online.", pod_name)
+
+        LOGGER.info("Step 3: Read-Verify already overwritten object in healthy cluster")
+        download_path = os.path.join(self.test_dir_path, f"{self.object_name}_download.txt")
+        resp = self.s3_test_obj.object_download(self.bucket_name, self.object_name, download_path)
+        assert_utils.assert_true(resp[0], f"Object download failed. Response: {resp[1]}")
+        dnld_checksum = self.ha_obj.cal_compare_checksum([download_path], compare=False)[0]
+        system_utils.remove_file(file_path=download_path)
+        assert_utils.assert_equal(upld_chcksm, dnld_checksum,
+                                  "Upload & download checksums doesn't match. Expected: "
+                                  f"{upld_chcksm} Actual: {dnld_checksum}")
+        LOGGER.info("Step 3: Successfully Read-Verify already overwritten object in healthy "
+                    "cluster")
+        new_s3_data = dict()
+        if CMN_CFG["dtm0_disabled"]:
+            LOGGER.info("Step 4: Start overwrite operation on new buckets")
```

Comment:
so if we create these buckets in healthy cluster we might not need DTM flag here, isn't it?

Reply:
if DTM is disabled, it will create a new bucket, otherwise it will use already created buckets.

Reply:
Updated